File: apps/api/data/mysql/stock_check_transformer.go
package mysql

import "apps/api/domain"

func ToStockCheckItemDB(d domain.StockCheckItem) StockCheckItem {
	return StockCheckItem{
		Id:               d.Id,
		StockCheckId:     d.StockCheckId,
		MaterialId:       d.MaterialId,
		CurrentStock:     d.CurrentStock,
		MaterialName:     d.MaterialName,
		Price:            d.Price,
		PurchaseUnit:     d.PurchaseUnit,
		PurchaseUnitSize: d.PurchaseUnitSize,
		MinimumStock:     d.MinimumStock,
		NormalStock:      d.NormalStock,
		CreatedAt:        d.CreatedAt,
	}
}

func ToStockCheckItemDomain(db StockCheckItem) domain.StockCheckItem {
	return domain.StockCheckItem{
		Id:               db.Id,
		StockCheckId:     db.StockCheckId,
		MaterialId:       db.MaterialId,
		CurrentStock:     db.CurrentStock,
		MaterialName:     db.MaterialName,
		Price:            db.Price,
		PurchaseUnit:     db.PurchaseUnit,
		PurchaseUnitSize: db.PurchaseUnitSize,
		MinimumStock:     db.MinimumStock,
		NormalStock:      db.NormalStock,
		CreatedAt:        db.CreatedAt,
	}
}

func ToStockCheckDB(d domain.StockCheck) StockCheck {
	items := make([]StockCheckItem, 0, len(d.Items))
	for _, item := range d.Items {
		items = append(items, ToStockCheckItemDB(item))
	}
	return StockCheck{
		Id:        d.Id,
		CreatedAt: d.CreatedAt,
		DeletedAt: d.DeletedAt,
		Items:     items,
	}
}

func ToStockCheckDomain(db StockCheck) domain.StockCheck {
	items := make([]domain.StockCheckItem, 0, len(db.Items))
	for _, item := range db.Items {
		items = append(items, ToStockCheckItemDomain(item))
	}
	return domain.StockCheck{
		Id:        db.Id,
		CreatedAt: db.CreatedAt,
		DeletedAt: db.DeletedAt,
		Items:     items,
	}
}

func ToStockCheckListDomain(dbs []StockCheck) []domain.StockCheck {
	result := make([]domain.StockCheck, 0, len(dbs))
	for _, db := range dbs {
		result = append(result, ToStockCheckDomain(db))
	}
	return result
}

File: libs/ui/src/data/api/stockCheck.transformer.ts
// eslint-disable-next-line @nx/enforce-module-boundaries
import {
  StockCheck as ApiStockCheck,
  StockCheckItem as ApiStockCheckItem,
} from '../../../../api-contract/src';
import { StockCheck, StockCheckForm, StockCheckItem } from '../../domain';

export function toStockCheckItem(item: ApiStockCheckItem): StockCheckItem {
  return {
    id: item.id,
    stockCheckId: item.stockCheckId,
    materialId: item.materialId,
    currentStock: item.currentStock,
    materialName: item.materialName,
    price: item.price,
    purchaseUnit: item.purchaseUnit,
    purchaseUnitSize: item.purchaseUnitSize,
    minimumStock: item.minimumStock,
    normalStock: item.normalStock,
    createdAt: item.createdAt,
  };
}

export function toStockCheck(stockCheck: ApiStockCheck): StockCheck {
  return {
    id: stockCheck.id,
    createdAt: stockCheck.createdAt,
    items: stockCheck.items.map(toStockCheckItem),
  };
}

export function toApiStockCheckForm(form: StockCheckForm) {
  return {
    items: form.items.map((item) => ({
      materialId: item.materialId,
      // zod validation guarantees a number by submission time; fall back to 0 to satisfy the API type.
      currentStock: item.currentStock ?? 0,
    })),
  };
}
